docs/data_types/table/table.md

```diff
@@ -0,0 +1,116 @@
+You can easily load such data from a given `.csv` file into a DocumentArray. 
+Let's take a look at the following example file, which includes data about book and their authors and publishing year.
+
+```text
+title,author,year
+Harry Potter and the Philosopher's Stone,J. K. Rowling,1997
+Klara and the sun,Kazuo Ishiguro,2020
+A little life,Hanya Yanagihara,2015
+```
+
+First we have to define the Document schema describing our data.
+```python
+from docarray import BaseDoc
+
+
+class Book(BaseDoc):
+    title: str
+    author: str
+    year: int
+```
+Next, we can load the content of the csv file to a DocList instance of `Book`s.
+```python
+from docarray import DocList
+
+
+docs = DocList[Book].from_csv(file_path='books.csv')
+docs.summary()
+```
+<details>
+    <summary>Output</summary>
+    ``` { .text .no-copy }
+    ╭────── DocList Summary ──────╮
+    │                             │
+    │   Type     DocList[Book]    │
+    │   Length   3                │
+    │                             │
+    ╰─────────────────────────────╯
+    ╭── Document Schema ──╮
+    │                     │
+    │   Book              │
+    │   ├── title: str    │
+    │   ├── author: str   │
+    │   └── year: int     │
+    │                     │
+    ╰─────────────────────╯
+    ```
+</details>
+
+The resulting DocList contains three `Book`s, since each row of the csv file corresponds to one book and is assigned to one `Book` instance.
+
+
+## Save to CSV file
+
+Vice versa, you can also store your DocList data to a `.csv` file.
+```python
+docs.to_csv(file_path='/path/to/my_file.csv')
+```
+
+Tabular data is often not the best choice to represent nested Documents. Hence, nested Documents will be stored flattened and can be accessed by their "__"-separated access paths.
+
+Let's look at an example. We now want to store not only the book data, but moreover book review data. Our `BookReview class` has a nested `book` attribute as well as the non-nested attributes `n_ratings` and `stars`.
+
+```python
+class BookReview(BaseDoc):
+    book: Book
+    n_ratings: int
+    stars: float
+
+
+review_docs = DocList[BookReview](
+    [BookReview(book=book, n_ratings=12345, stars=5) for book in docs]
+)
+review_docs.summary()
+```
+<details>
+    <summary>Output</summary>
+    ``` { .text .no-copy}
+    ╭───────── DocList Summary ─────────╮
+    │                                   │
+    │   Type     DocList[BookReview]    │
+    │   Length   3                      │
+    │                                   │
+    ╰───────────────────────────────────╯
+    ╭──── Document Schema ────╮
+    │                         │
+    │   BookReview            │
+    │   ├── book: Book        │
+    │   │   ├── title: str    │
+    │   │   ├── author: str   │
+    │   │   └── year: int     │
+    │   ├── n_ratings: int    │
+    │   └── stars: float      │
+    │                         │
+    ╰─────────────────────────╯
+    ```
+</details>
+
```

Comment:
we should either explain the options that `from_csv` and `to_csv` have, or link to their docstrings